colobot-base/src/common/settings.cpp

```diff
@@ -269,7 +275,9 @@ void CSettings::LoadSettings()
 
     CInput::GetInstancePointer()->LoadKeyBindings();
 
-
+    std::deque<std::string> commands;
+    GetConfigFile().GetDequeProperty("Setup", "CommandHistory", commands);
+    main->SetCommandHistory(commands);
```

Comment:
SetCommandHistory() takes ownership of the memory instead of creating a copy. Someone could later introduce a bug by expecting it to create a copy. Use `std::deque<std::string>&&` in SetCommandHistory and use std::move here to make the ownership transfer more obvious.